src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -20410,9 +20427,9 @@ void Unit::RewardRage(uint32 damage, uint32 weaponSpeedHitFactor, bool attacker)
 
     if (attacker)
     {
-        addRage = (damage / rageconversion * 7.5f + weaponSpeedHitFactor) / 2;
-
-        // talent who gave more rage on attack
+        float rageFromDamageDealt = damage / rageconversion * 7.5f;
+        addRage = (rageFromDamageDealt + weaponSpeedHitFactor) / 2.0f;
+        addRage = std::min(addRage, rageFromDamageDealt * 2.0f);
```

Comment:
Im just interested in knowing where the calculation comes from

Reply:
formula sources are linked in source^  wiki source is easier to read and takes formulas from bluepost

![image](https://github.com/user-attachments/assets/b95a3c34-aee5-490c-9547-a25aee4c18bc)

implementation here:
weaponSpeedHitFactor = `f*s`
damage = `d`
rageconversion = `c`

right hand side `<=` is a cap on this value to avoid low damage hits being averaged up by weaponspeed factor, bluepost explains it best

Reply:
Looks good to me, I would be more compliant with the formula for improving the readability and I would leave a comment mentioning the link fo the formula.
```cpp
        // Rage formula https://wowwiki-archive.fandom.com/wiki/Rage#Formulae
        float rageFromDamageDealt = (damage / rageconversion) * 15.0f;
        addRage = ((rageFromDamageDealt/2) + weaponSpeedHitFactor) / 2.0f;
        addRage = std::min(addRage, rageFromDamageDealt);
```

or, you could leave as it is, but at least put the formula in the comment

Reply:
The formula lines match 1 to 1 the [bluepost linked in sources](https://web.archive.org/web/20090604123729/http://forums.worldofwarcraft.com/thread.html?topicId=17367760070&pageNo=13#), the wiki mentions this post but the link is broken. 
```
    1) Determine rage based on damage done divided by a level constant (7.5*d/c)
    2) Determine how much rage would be 3.5 rage per second when factoring in the hit made and the weapon’s speed (f*s)
    3) Average the two of them together (the divided by 2 part).
```

I updated the function to include the wiki link